include/sound/sof/pm.h

```diff
@@ -45,4 +45,9 @@ struct sof_ipc_pm_core_config {
 	uint32_t enable_mask;
 } __packed;
 
+struct sof_ipc_pm_gate {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t flags;
+} __packed;
```

Comment:
don't we usually add reserved fields for future expansion?

Reply:
also you need to add a comment that the value for this flags field is platform-dependent, and this PM_GATE IPC will only be called from platform-specific layers, not by the SOF core.

Reply:
We have reserved bits in the flags, and actually we are considering adding a new bit in flags for DMA trace disabling indication.

Reply:
@keyonjie better to add a few reserved fields. Who knows how the pm gate ipc will evolve for future platforms?

Reply:
@ranj063 @plbossart because this is a compact IPC, what payload we can transfer should be <=32 bits, so what we can reserve is within the uint32_t flags only. e.g. 
```
#define SOF_PM_PG_RSVD		BIT(0)

/* Indicates whether streaming is active */
#define SOF_PM_PG_STREAMING	BIT(1)

/* Prevent power gating (0 - deep power state transitions allowed) */
#define SOF_PM_PPG		BIT(2)

/* Prevent clock gating (0 - cg allowed, 1 - DSP clock always on) */
#define SOF_PM_PCG		BIT(3)

/* Disable DMA tracing (0 - keep tracing, 1 - to disable DMA trace) */
#define SOF_PM_NO_TRACE		BIT(4)
```

Reply:
> @ranj063 @plbossart because this is a compact IPC, what payload we can transfer should be <=32 bits, so what we can reserve is within the uint32_t flags only. e.g.

No. you are defining an IPC, and it can be extended on other platforms.
The fact that the IPC is compact is an Intel-specific feature/problem, and you are not using the usual IPC mechanism to transmit the data.

Reply:
OK, it makes sense from this angle, @plbossart let me add reserved fields for it.